System.IO.Abstractions.TestingHelpers/MockFile.cs

```diff
@@ -37,6 +39,28 @@ public override void AppendAllLines(string path, IEnumerable<string> contents, E
             AppendAllText(path, concatContents, encoding);
         }
 
+#if NETCOREAPP2_0
+        public async override Task AppendAllLinesAsync(string path, IEnumerable<string> contents, CancellationToken cancellationToken = default)
+        {
+            mockFileDataAccessor.PathVerifier.IsLegalAbsoluteOrRelative(path, "path");
+            VerifyValueIsNotNull(contents, "contents");
+
+            await AppendAllLinesAsync(path, contents, MockFileData.DefaultEncoding, cancellationToken);
```

Comment:
A couple of comments regarding the async implementation here:
1. Why are you awaiting this method instead of returning directly `return AppendAllLinesAsync(...);`?
2. For performance reasons any method that doesn't contain an `await` shouldn't be marked with the `async` keyword. Marking a method with the `async` keyword adds overhead and slows things down.

Reply:
This comment applies to the rest of the async methods in this PR as well, not just here.

Reply:
To answer your question, it was just oversight on my part. I'll clean those up.